crates/exex/exex/src/dyn_context.rs

```diff
@@ -0,0 +1,56 @@
+//! Mirrored version of [`ExExContext`](`crate::ExExContext`)
+//! without generic abstraction over [<Node>](`reth_node_api::FullNodeComponents`)
+
+use std::sync::Arc;
+
+use reth_chainspec::{EthChainSpec, Head};
+use reth_node_api::FullNodeComponents;
+use reth_node_core::node_config::NodeConfig;
+use tokio::sync::mpsc;
+
+use crate::{ExExContext, ExExEvent};
+
+// TODO(0xurb) - add `node` and `notifications` after abstractions
+/// Captures the context that an `ExEx` has access to.
+#[derive(Debug)]
+pub struct ExExContextDyn {
+    /// The current head of the blockchain at launch.
+    pub head: Head,
+    /// The config of the node
+    pub config: NodeConfig<Box<dyn EthChainSpec + 'static>>,
+    /// The loaded node config
+    pub reth_config: reth_config::Config,
+    /// Channel used to send [`ExExEvent`]s to the rest of the node.
+    ///
+    /// # Important
+    ///
+    /// The exex should emit a `FinishedHeight` whenever a processed block is safe to prune.
+    /// Additionally, the exex can pre-emptively emit a `FinishedHeight` event to specify what
+    /// blocks to receive notifications for.
+    pub events: mpsc::UnboundedSender<ExExEvent>,
+}
+
+impl<Node: FullNodeComponents> From<ExExContext<Node>> for ExExContextDyn {
+    fn from(ctx: ExExContext<Node>) -> Self {
+        // convert `NodeConfig` with generic over chainspec into `NodeConfig<Box<dyn EthChainSpec>`
+        let chain: Arc<Box<dyn EthChainSpec + 'static>> =
+            Arc::new(Box::new(ctx.config.chain) as Box<dyn EthChainSpec>);
+        let config = NodeConfig {
```

Comment:
looks like we could use this mapping as a function on NodeConfig directly

Reply:
I don't think it belongs in node builder crates https://github.com/paradigmxyz/reth/pull/11983#discussion_r1812411496

Reply:
what I meant was a new function on NodeConfig that does this

https://github.com/paradigmxyz/reth/blob/1f5a3dbcc79f8ab0f5831046c1144f58f2ed8e19/crates/node/core/src/node_config.rs#L141-L141

but we can also move this later

Reply:
oh right, I see. I agree, we can introduce a method on `NodeConfig` for replacing the chainspec

Reply:
@shekhirin should I fill **this** PR with 

```rust
impl<ChainSpec> NodeConfig<ChainSpec> { // some mapping to  NodeConfig<Box<dyn ...>>  }
```

Reply:
it's fine to do in another PR, can you please create an issue to not forget?

Reply:
@shekhirin https://github.com/paradigmxyz/reth/issues/12040
maybe good-first-issue tho?